Main.playground/Contents.swift

```diff
@@ -6,28 +6,129 @@
 //
 import Foundation
 
-class Node{
+class Node: Hashable, Equatable , Comparable{
     var id: UUID
     var description: String
     init(){
         self.id = UUID()
         self.description = "this is a description for \(self.id)"
     }
+    static func == (lhs: Node, rhs: Node) -> Bool {
+        if lhs.id == rhs.id && lhs.description == rhs.description{
+            return true
+        }
+        return false
+    }
+    func hash(into hasher: inout Hasher) {
+        hasher.combine(id)
+        hasher.combine(description)
+    }
+    static func < (lhs: Node, rhs: Node) -> Bool {
+        return lhs.id.uuidString < rhs.id.uuidString
+    }
+    
 }
-
-class Stack<T>{
+class ExtraOperations<T: Hashable & Comparable>{
     var array: [T]
+    var maxCapacity: Int?
```

Comment:
how is it an optional and it's required in the init??

something needs to be changed depending on the requirments